forespoersel-marker-besvart/src/main/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/forespoerselmarkerbesvart/MarkerForespoerselBesvartRiver.kt

```diff
@@ -0,0 +1,89 @@
+package no.nav.helsearbeidsgiver.inntektsmelding.forespoerselmarkerbesvart
+
+import no.nav.helse.rapids_rivers.JsonMessage
+import no.nav.helse.rapids_rivers.MessageContext
+import no.nav.helse.rapids_rivers.RapidsConnection
+import no.nav.helse.rapids_rivers.River
+import no.nav.helsearbeidsgiver.felles.EventName
+import no.nav.helsearbeidsgiver.felles.Key
+import no.nav.helsearbeidsgiver.felles.json.les
+import no.nav.helsearbeidsgiver.felles.json.toMap
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.demandValues
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.pritopic.Pri
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.pritopic.PriProducer
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.rejectKeys
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.require
+import no.nav.helsearbeidsgiver.felles.utils.Log
+import no.nav.helsearbeidsgiver.utils.json.fromJson
+import no.nav.helsearbeidsgiver.utils.json.parseJson
+import no.nav.helsearbeidsgiver.utils.json.serializer.UuidSerializer
+import no.nav.helsearbeidsgiver.utils.json.toJson
+import no.nav.helsearbeidsgiver.utils.json.toPretty
+import no.nav.helsearbeidsgiver.utils.log.MdcUtils
+import no.nav.helsearbeidsgiver.utils.log.logger
+import no.nav.helsearbeidsgiver.utils.log.sikkerLogger
+import java.util.UUID
+
+class MarkerForespoerselBesvartRiver(
+    rapid: RapidsConnection,
+    private val priProducer: PriProducer
+) : River.PacketListener {
+
+    private val logger = logger()
+    private val sikkerLogger = sikkerLogger()
+
+    init {
+        River(rapid).validate { msg ->
+            msg.demandValues(
+                Key.EVENT_NAME to EventName.INNTEKTSMELDING_MOTTATT.name
+            )
+            msg.require(
+                Key.UUID to { it.fromJson(UuidSerializer) },
+                Key.FORESPOERSEL_ID to { it.fromJson(UuidSerializer) }
+            )
+            msg.rejectKeys(
+                Key.BEHOV,
+                Key.DATA
+            )
+        }
+            .register(this)
+    }
+
+    override fun onPacket(packet: JsonMessage, context: MessageContext) {
+        val json = packet.toJson().parseJson()
```

Comment:
Litt flisespikk, og mulig det kanskje heller smeller i init validate, men en try catch blokk for å unngå at app kræsjer på poison pills bør kanskje være standard for alle våre R&R-apper, i hvert fall i fremtiden? 
Mulig vi skal implementere en Dead Letter Queue før vi gjør dette overalt uansett.

Reply:
Det burde det, men problemet er at vi ikke har helt kontroll over poison pill-problematikken når vi bruker RR. Under panseret så gjør RR noe prosessering av json, som er sårbart for poison pill, som vi ikke kan endre. Når vi først har kommet til en `onPacket` så kan vi være (ganske) sikre på at json-en er gyldig.